server/routerlicious/packages/services-telemetry/src/lumber.ts

```diff
@@ -0,0 +1,149 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { LumberEventName } from "./lumberEventNames";
+import { LogLevel, LumberType, ITelemetryMetadata, ILumberjackEngine } from "./resources";
+
+// Lumber represents the telemetry data being captured, and it uses a list of
+// ILumberjackEngine to emit the data according to the engine implementation.
+// Lumber should be created through Lumberjack. Additional properties can be added through
+// addProperty(). Once the telemetry event is complete, the user must call either success()
+// or error() on Lumber to emit the data.
+export class Lumber<T extends string = LumberEventName> {
+    private readonly _eventName: T;
+    private readonly _properties: Map<string, any>;
+    private readonly _timestamp: string;
+    private readonly startTime: number;
+    private readonly _type: LumberType;
```

Comment:
nit: `private readonly` with an accompanying `public get` is more readable as simply `public readonly`